File: app/src/main/java/com/metahumanz/pacilread/model/ReadingTimeEntryRecord.java
package com.metahumanz.pacilread.model;

import org.json.JSONObject;

public class ReadingTimeEntryRecord {
    public long id;
    public String date;
    public String sourceDeviceId;
    public String bookIdentity;
    public String bookTitle;
    public String bookAuthor;
    public int durationSeconds;
    public int charCount;
    public long updatedAt;

    public JSONObject toJson() {
        JSONObject json = new JSONObject();
        try {
            json.put("id", id);
            json.put("date", date != null ? date : "");
            json.put("sourceDeviceId", sourceDeviceId != null ? sourceDeviceId : "");
            json.put("bookIdentity", bookIdentity != null ? bookIdentity : "");
            json.put("bookTitle", bookTitle != null ? bookTitle : "");
            json.put("bookAuthor", bookAuthor != null ? bookAuthor : "");
            json.put("durationSeconds", durationSeconds);
            json.put("charCount", charCount);
            json.put("updatedAt", updatedAt);
        } catch (Exception ignore) {}
        return json;
    }

    public static ReadingTimeEntryRecord fromJson(JSONObject json) {
        ReadingTimeEntryRecord entry = new ReadingTimeEntryRecord();
        entry.id = json.optLong("id", 0);
        entry.date = json.optString("date", "");
        entry.sourceDeviceId = json.optString("sourceDeviceId", "");
        entry.bookIdentity = json.optString("bookIdentity", "");
        entry.bookTitle = json.optString("bookTitle", "");
        entry.bookAuthor = json.optString("bookAuthor", "");
        entry.durationSeconds = json.optInt("durationSeconds", 0);
        entry.charCount = json.optInt("charCount", 0);
        entry.updatedAt = json.optLong("updatedAt", 0);
        return entry;
    }
}

File: app/src/main/kotlin/com/metahumanz/pacilread/model/ReadingTimeEntryRecord.kt
package com.metahumanz.pacilread.model

import org.json.JSONObject

class ReadingTimeEntryRecord {
    @JvmField var id: Long = 0
    @JvmField var date: String? = null
    @JvmField var sourceDeviceId: String? = null
    @JvmField var bookIdentity: String? = null
    @JvmField var bookTitle: String? = null
    @JvmField var bookAuthor: String? = null
    @JvmField var durationSeconds: Int = 0
    @JvmField var charCount: Int = 0
    @JvmField var updatedAt: Long = 0

    fun toJson(): JSONObject {
        val json = JSONObject()
        try {
            json.put("id", id)
            json.put("date", date ?: "")
            json.put("sourceDeviceId", sourceDeviceId ?: "")
            json.put("bookIdentity", bookIdentity ?: "")
            json.put("bookTitle", bookTitle ?: "")
            json.put("bookAuthor", bookAuthor ?: "")
            json.put("durationSeconds", durationSeconds)
            json.put("charCount", charCount)
            json.put("updatedAt", updatedAt)
        } catch (_: Exception) {
        }
        return json
    }

    companion object {
        @JvmStatic
        fun fromJson(json: JSONObject): ReadingTimeEntryRecord = ReadingTimeEntryRecord().apply {
            id = json.optLong("id", 0)
            date = json.optString("date", "")
            sourceDeviceId = json.optString("sourceDeviceId", "")
            bookIdentity = json.optString("bookIdentity", "")
            bookTitle = json.optString("bookTitle", "")
            bookAuthor = json.optString("bookAuthor", "")
            durationSeconds = json.optInt("durationSeconds", 0)
            charCount = json.optInt("charCount", 0)
            updatedAt = json.optLong("updatedAt", 0)
        }
    }
}
